File: src/main/java/imaxct/handler/MyExceptionHandler.java
package imaxct.handler;

import imaxct.bean.Msg;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.ResponseBody;

import java.sql.SQLException;

/**
 * Created by imaxct on 17-4-25.
 * student
 */
@ControllerAdvice
public class MyExceptionHandler {

    @ResponseBody
    @ExceptionHandler(Exception.class)
    public Msg handleAllException(Exception e){
        System.out.println(e.getClass());
        e.printStackTrace();
        return new Msg("Access Denied.");
    }

    @ResponseBody
    @ExceptionHandler(SQLException.class)
    public Msg handleSQLException(){
        return new Msg("系统错误, 请求失败.");
    }

}

File: src/main/java/imaxct/handler/MyExceptionHandler.kt
package imaxct.handler

import imaxct.bean.Msg
import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.ResponseBody

import java.sql.SQLException

/**
 * Created by imaxct on 17-4-25.
 * student
 */
@ControllerAdvice
class MyExceptionHandler {

    @ResponseBody
    @ExceptionHandler(Exception::class)
    fun handleAllException(e: Exception): Msg<*> {
        e.printStackTrace()
        return Msg<Int>("系统忙, 请稍候")
    }

    @ResponseBody
    @ExceptionHandler(SQLException::class)
    fun handleSQLException(): Msg<*> = Msg<Int>("系统错误, 请求失败.")

}